src/CourseDashboard.hpp

```diff
@@ -1,19 +1,11 @@
 #pragma once
-#include "User.hpp"
-#include <vector>
-
+#include "UserData.hpp"
 
 class CourseDashboard
 {
+
 public:
     CourseDashboard();
-    void showAll();
-    void createUser(const User & user);
-    void deleteUserByNick(std::string nick);
-    void retriveUserByNick(std::string nick);
-    void updateUser(User & user);
-
-private:
-    std::vector<User> users_;
```

Comment:
You have moved this code into UserData, which is ok, but the UserData class be here a part of CourseDashboard class. We need an access to this data.